common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -215,36 +213,19 @@ bool base_wlan_hal_dwpal::fsm_setup()
                         }
                     }
                 } else { // Auto Channel
-                    if (get_hal_conf().ap_passive_mode_enabled) {
-                        if (m_radio_info.wifi_ctrl_enabled == 1) {
-                            if (m_radio_info.tx_enabled) {
-                                LOG(DEBUG) << "ap_enabled tx = " << int(m_radio_info.tx_enabled);
-                                return true;
-                            } else {
-                                LOG(ERROR) << "ap_enabled with tx OFF!";
-                                error = true;
-                            }
-                        } else if (m_radio_info.wifi_ctrl_enabled == 2) {
-                            LOG(ERROR)
-                                << "Auto channel with passive_mode=on, wifi_ctrl_enabled = 2";
-                            error = true;
-                        }
-                    } else { // ap_passive_mode_enabled == false
-
-                        if (m_radio_info.wifi_ctrl_enabled == 2) {
-
-                            if (m_radio_info.tx_enabled) {
-                                LOG(DEBUG) << "ap_enabled tx = " << int(m_radio_info.tx_enabled);
-                                return true;
-                            } else {
-                                LOG(ERROR) << "ap_enabled with tx OFF!";
-                                error = true;
-                            }
-                        } else if (m_radio_info.wifi_ctrl_enabled == 1) {
-                            LOG(ERROR)
-                                << "Auto channel with passive_mode=off, wifi_ctrl_enabled = 1";
+                    if (m_radio_info.wifi_ctrl_enabled == 1) {
+                        if (m_radio_info.tx_enabled) {
+                            LOG(DEBUG) << "ap_enabled tx = " << int(m_radio_info.tx_enabled);
+                            return true;
+                        } else {
+                            LOG(ERROR) << "ap_enabled with tx OFF!";
                             error = true;
                         }
+                    } else if (m_radio_info.wifi_ctrl_enabled != 0) {
```

Comment:
Maybe we should change wifi_ctrl_enabled into a bool parameter then? (nitpick)

Reply:
We should change its name. It is basically `hostapd_cli -iwlanX status`.. the name is super confusing. We used it for the `ACS_DONE` patch in hostapd in which case `wifi_ctrl_enabled=2` after hostapd init complete, and beerocks sent `chann_switch` command which triggered hostapd init completion. The whole point was to make sure no clients ever connect before beerocks was attached, so we won't miss connected clients.
In passive mode we simply don't need this anymore (we get the connected clients in a different way).

Bottom line - we should change this to `bool hostap_enabled` and I don't think it should be a (nitpick), but a requirement. But this can be done in a separate PR..

Reply:
I kept it and integer because I didn't know if the hostapd is still able to send `2`.